stdlib/universe/holt_winters/neldermead.go

```diff
@@ -0,0 +1,238 @@
+package holt_winters
+
+import "math"
+
+const (
+	defaultMaxIterations = 1000
+	// reflection coefficient
+	defaultAlpha = 1.0
+	// contraction coefficient
+	defaultBeta = 0.5
+	// expansion coefficient
+	defaultGamma = 2.0
+)
+
+// Optimizer is an implementation of the Nelder-Mead optimization method.
+// Based on work by Michael F. Hutt: http://www.mikehutt.com/neldermead.html
+type Optimizer struct {
+	// Maximum number of iterations.
+	MaxIterations int
+	// Reflection coefficient.
+	Alpha,
+	// Contraction coefficient.
+	Beta,
+	// Expansion coefficient.
+	Gamma float64
+}
+
+// NewOptimizer returns a new instance of Optimizer with all values set to the defaults.
+func NewOptimizer() *Optimizer {
+	return &Optimizer{
+		MaxIterations: defaultMaxIterations,
+		Alpha:         defaultAlpha,
+		Beta:          defaultBeta,
+		Gamma:         defaultGamma,
+	}
+}
+
+// Optimize applies the Nelder-Mead simplex method with the Optimizer's settings.
+func (o *Optimizer) Optimize(
+	objfunc func([]float64) float64,
+	start []float64,
+	epsilon,
+	scale float64,
+) (float64, []float64) {
+	n := len(start)
```

Comment:
n is very small, right?  I think slices are much more efficient in this case.

Reply:
See above